protocol/messenger_storenode_request_test.go

```diff
@@ -0,0 +1,143 @@
+package protocol
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/suite"
+	"go.uber.org/zap"
+
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/status-im/status-go/appdatabase"
+	gethbridge "github.com/status-im/status-go/eth-node/bridge/geth"
+	"github.com/status-im/status-go/eth-node/types"
+	"github.com/status-im/status-go/params"
+	"github.com/status-im/status-go/protocol/protobuf"
+	"github.com/status-im/status-go/protocol/requests"
+	"github.com/status-im/status-go/protocol/tt"
+	"github.com/status-im/status-go/t/helpers"
+
+	mailserversDB "github.com/status-im/status-go/services/mailservers"
+	waku2 "github.com/status-im/status-go/wakuv2"
+)
+
+const (
+	localFleet        = "local-test-fleet-1"
+	localMailserverID = "local-test-mailserver"
+)
+
+func TestMessengerStoreNodeRequestSuite(t *testing.T) {
+	suite.Run(t, new(MessengerStoreNodeRequestSuite))
+}
+
+type MessengerStoreNodeRequestSuite struct {
+	suite.Suite
+
+	owner *Messenger
+	bob   *Messenger
+
+	wakuStoreNode *waku2.Waku
+	ownerWaku     types.Waku
+	bobWaku       types.Waku
+
+	logger *zap.Logger
+}
+
+func (s *MessengerStoreNodeRequestSuite) newMessenger(shh types.Waku, logger *zap.Logger, mailserverAddress string) *Messenger {
+	privateKey, err := crypto.GenerateKey()
+	s.Require().NoError(err)
+
+	mailserversSQLDb, err := helpers.SetupTestMemorySQLDB(appdatabase.DbInitializer{})
+	s.Require().NoError(err)
+
+	mailserversDatabase := mailserversDB.NewDB(mailserversSQLDb)
+	err = mailserversDatabase.Add(mailserversDB.Mailserver{
+		ID:      localMailserverID,
+		Name:    localMailserverID,
+		Address: mailserverAddress,
+		Fleet:   localFleet,
+	})
+	s.Require().NoError(err)
+
+	options := []Option{
+		WithMailserversDatabase(mailserversDatabase),
+		WithClusterConfig(params.ClusterConfig{
+			Fleet: localFleet,
+		}),
+	}
+
+	messenger, err := newMessengerWithKey(shh, privateKey, logger, options)
+
+	s.Require().NoError(err)
+	return messenger
+}
+
+func (s *MessengerStoreNodeRequestSuite) SetupTest() {
+	s.logger = tt.MustCreateTestLogger()
+
+	// Create store node
+
+	storeNodeLogger := s.logger.With(zap.String("name", "store-node-waku"))
+	s.wakuStoreNode = NewWakuV2(&s.Suite, storeNodeLogger, true, true)
+
+	//protocols := s.wakuStoreNode.Protocols()
+	//s.Require().Contains(protocols, store.StoreID_v20beta4)
```

Comment:
leftover

Reply:
oh, this is actually a question that I forgot to ask @richard-ramos!
Is there a way to check that the waku node successfully supports store protocol?

Reply:
ok, Richard proposed some options, but this seems to be redundant to do.
I'll remove these lines in upcoming PRs, not to rerun tests here.